File: Lab 4/Lab 4 Java/Gerente.java

public class Gerente extends Funcionario {
    private int numeroEquipes;

    public Gerente(String nome, int salario, int numeroEquipes) {
        super(nome, salario);
        this.numeroEquipes = numeroEquipes;
    }

    @Override
    public String descricao() {
        return super.descricao() + "\nNumero de Equipes: " + numeroEquipes;
    }
}

File: Lab 4/Lab 4 c++/Gerente.cpp

#include "Gerente.h"

Gerente::Gerente(const std::string& nome, int salario, int numeroEquipes)
    : Funcionario(nome, salario), numeroEquipes(numeroEquipes) {}

int Gerente::getNumeroEquipes() const {
    return numeroEquipes;
}

void Gerente::setNumeroEquipes(int numeroEquipes) {
    this->numeroEquipes = numeroEquipes;
}

int Gerente::bonus() const {
    return numeroEquipes * 500;
}

std::string Gerente::descricaoGerente() const {
    return "\nNumero de equipes: " + std::to_string(numeroEquipes);
}

std::string Gerente::descricao() const {
    return Funcionario::descricao() + descricaoGerente();
}
